packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -388,6 +389,43 @@ class RelativePositionedTransition extends AnimatedWidget {
   }
 }
 
+/// Animated version of a [DecoratedBox] that animates the different properties
+/// of its [Decoration].
+class DecoratedBoxTransition extends AnimatedWidget {
+  /// Creates an animated [DecorationBox] whose [Decoration] animation updates
+  /// the widget. 
+  /// 
+  /// `decoration` and `position` cannot be null.
+  /// 
+  /// Also see [DecoratedBox].
```

Comment:
I actually meant for the user to read DecoratedBox's constructor for additional constraints but I guess there's no way to refer to the constructor specifically